test/common/http/http1/conn_pool_test.cc

```diff
@@ -540,5 +541,41 @@ TEST_F(Http1ConnPoolImplTest, DrainCallback) {
   dispatcher_.clearDeferredDeleteList();
 }
 
+TEST_F(Http1ConnPoolImplTest, RemoteClose) {
```

Comment:
RemoteClose is not a good test name. It's not really what you are testing. Here is what you are testing:
1) Send a request on the conn pool
2) Receive response headers
3) Receive some response body
4) Upstream closes connection to indicate that response is finished.

In old code this causes a crash. Please rename test case appropriately and also clearly put description of what causes the crash in the PR description.